lib/util/db.js

```diff
@@ -239,7 +241,65 @@ const queryFuncs = (db, obj) => {
     for (let i = 0; i < xs.length; i += 1) result[i] = f(xs[i]);
     return result;
   };
-  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, resolve).catch(reject1));
+
+  obj.stream = (s) => new Promise((resolve, reject1) => db.stream(s, stream => {
+    try {
```

Comment:
What below might throw an error that would be caught here? `stream.pipe()`? Attaching one of the listeners? Or is this more of a safeguard?

Reply:
With all the surprising error handling required for streams, I wanted to make sure that _any_ errors were caught, even if they don't seem likely.